projects/plugins/jetpack/modules/contact-form/admin.php

```diff
@@ -1128,3 +1129,128 @@ function grunion_feedback_admin_notice() {
 	}
 }
 add_action( 'admin_notices', 'grunion_feedback_admin_notice' );
+
+add_action( 'admin_enqueue_scripts', array( 'Grunion_Admin', 'maybe_enable_gdrive_export_button' ) );
+add_action( 'wp_ajax_grunion_export_to_gdrive', array( 'Grunion_Admin', 'export_to_gdrive' ) );
+
+/**
+ * Class Grunion_Admin
+ *
+ * Stop the global scope madness, use static methods on this class to provide for wp-admin responses area.
+ * Might turn into singleton in the future.
+ */
+class Grunion_Admin {
+	/**
+	 * Exports data to Google Drive, based on POST data.
+	 *
+	 * @see Grunion_Contact_Form_Plugin::get_feedback_entries_from_post
+	 */
+	public static function export_to_gdrive() {
+		$grunion     = Grunion_Contact_Form_Plugin::init();
+		$export_data = $grunion->get_feedback_entries_from_post();
+
+		$fields    = array_keys( $export_data );
+		$row_count = count( reset( $export_data ) );
+
+		$sheet_data = array( $fields );
+
+		for ( $i = 0; $i < $row_count; $i ++ ) {
+
+			$current_row = array();
+
+			/**
+			 * Put all the fields in `$current_row` array.
+			 */
+			foreach ( $fields as $single_field_name ) {
+				$current_row[] = $export_data[ $single_field_name ][ $i ];
+			}
+
+			$sheet_data[] = $current_row;
+		}
+
+		$user_id           = (int) get_current_user_id();
+		$spreadsheet_title = sprintf( '%s - %s', __( 'Responses', 'jetpack' ), gmdate( 'Y-m-d H:i' ) );
+		require_once JETPACK__PLUGIN_DIR . '_inc/lib/class-jetpack-google-drive-helper.php';
+		$sheet_response = Jetpack_Google_Drive_Helper::create_sheet( $user_id, $spreadsheet_title, $sheet_data );
```

Comment:
Nit: let's just call it `$sheet` - I believe it contains all the data we need on the response, but I would not want what this function returns to be considered a response in general.